api/reviewer_api/services/documentpageflagservice.py

```diff
@@ -172,6 +173,26 @@ def updatepageflags(
             pageflags, deldocpagesmapping, requestid, userinfo, redactionlayerid
         )
 
+    def updatepageflags_redactions_remaining(self, requestid, data, redactionlayerid, userinfo):
+        docids = [docflagobj["docid"] for docflagobj in data]
+        previousdocpageflag = self.getdocumentpageflagsbydocids(requestid, redactionlayerid, docids)
+        # loop through data and adjust docobj in data (updated page flags to apply to page / doc id) with previous docpageflag data (data from DocumentPageFlag Table)
+        for docflagobj in data:
+            for prevdocflagobj in previousdocpageflag:
+                if (docflagobj['docid'] == prevdocflagobj['documentid']):
+                    docflagobj['version'] = prevdocflagobj['documentversion']
+                    docflagobj["attributes"] = prevdocflagobj["attributes"]
+                    docflagobj['pageflag'] = [docflagobj['pageflag'] if docflagobj['pageflag']['page'] == pageflag['page'] else pageflag for pageflag in prevdocflagobj['pageflag']]
+            DocumentPageflag.savepageflag(
+                requestid,
+                docflagobj["docid"],
+                docflagobj["version"],
+                json.dumps(docflagobj['pageflag']),
+                json.dumps(userinfo),
+                redactionlayerid,
+                json.dumps(docflagobj["attributes"]),
+            )
+
```

Comment:
try to do a bulk save instead of saving this in a loop.

Reply:
Done - thanks for the feedback!